arcgis-ios-sdk-samples/Layers/Add ENC exchange set/AddENCExchangeSetViewController.swift

```diff
@@ -0,0 +1,108 @@
+// Copyright 2020 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class AddENCExchangeSetViewController: UIViewController {
+    /// The map view managed by the view controller.
+    @IBOutlet weak var mapView: AGSMapView! {
+        didSet {
+            mapView.map = AGSMap(basemap: .oceans())
+        }
+    }
+    
+    /// Load the ENC dataset and add to the map view.
+    ///
+    /// - Parameter mapView: The map view managed by the view controller.
+    func addENCExchangeSet(mapView: AGSMapView) {
+        guard let map = mapView.map else { return }
+        // Load catalog file in ENC exchange set from bundle.
+        let catalogURL = Bundle.main.url(
+            forResource: "CATALOG",
+            withExtension: "031",
+            subdirectory: "ExchangeSetwithoutUpdates/ExchangeSetwithoutUpdates/ENC_ROOT"
+        )!
+        let encExchangeSet = AGSENCExchangeSet(fileURLs: [catalogURL])
+        
+        // URL to the "hydrography" data folder that contains the "S57DataDictionary.xml" file.
+        let hydrographyDirectory = Bundle.main.url(
+            forResource: "S57DataDictionary",
+            withExtension: "xml",
+            subdirectory: "hydrography/hydrography"
+        )!
+        .deletingLastPathComponent()
+        // Set environment settings for loading the dataset.
+        let environmentSettings = AGSENCEnvironmentSettings.shared()
+        environmentSettings.resourceDirectory = hydrographyDirectory
+        // The SENC data directory is for temporarily storing generated files.
+        environmentSettings.sencDataDirectory = FileManager.default.temporaryDirectory
```

Comment:
It would be a good idea to create a subdirectory inside the temporary directory (see [this](https://nshipster.com/temporary-files/) NSHipster article for advice on how to do that). It would also be good to delete the temporary directory when the sample is closed, which would probably involve storing the temporary directory URL and then removing the item at the URL when the view controller is deinited.